src/Microsoft.Azure.SignalR.Common/ServiceConnections/MultiEndpointMessageWriter.cs

```diff
@@ -52,11 +53,55 @@ public Task WriteAsync(ServiceMessage serviceMessage)
             return WriteMultiEndpointMessageAsync(serviceMessage, connection => connection.WriteAsync(serviceMessage));
         }
 
-        public async Task<bool> WriteAckableMessageAsync(ServiceMessage serviceMessage, CancellationToken cancellationToken = default)
+        public Task<bool> WriteAckableMessageAsync(ServiceMessage serviceMessage, CancellationToken cancellationToken = default)
+        {
+            if (serviceMessage is CheckConnectionExistenceWithAckMessage)
+            {
+                return WriteSingleResultAckableMessage(serviceMessage, cancellationToken);
+            }
+            else
+            {
+                return WriteMultiResultAckableMessage(serviceMessage, cancellationToken);
+            }
+        }
+
+        /// <summary>
+        /// For user or group related operations, different endpoints might return different results
+        /// Strategy:
+        /// Always wait until all endpoints return or throw
+        /// 1. When any endpoint returns true, return true
+        /// 2. When all endpoints return false, return false
+        /// 3. When any endpoint throws, throw
+        /// </summary>
+        /// <param name="serviceMessage"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        private async Task<bool> WriteMultiResultAckableMessage(ServiceMessage serviceMessage, CancellationToken cancellationToken = default)
+        {
+            var tcs = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
+
+            var dict = new ConcurrentBag<bool>();
+            await WriteMultiEndpointMessageAsync(serviceMessage, async connection =>
+            {
+                dict.Add(await connection.WriteAckableMessageAsync(serviceMessage.Clone(), cancellationToken));
+            });
+
+            return dict.All(i => i);
+        }
+
+        /// <summary>
+        /// For connection related operations, since connectionId is globally unique, only one endpoint can have the connection
+        /// Strategy:
+        /// Don't need to wait until all endpoints return or throw
+        /// 1. Whenever any endpoint returns true: return true
+        /// 2. When all endpoints return false, return false
```

Comment:
Update the summary to match the implementation. The second step is to throw ex if any endpoint throw. And the third part is return false